File: src/ProjectR.Scripting/content/scripts/spells/lua/FlyingPhoenix.lua
function GetTargetType()
    return Single
end

function GetName()
return "Flying Phoenix"
end

function GetDescription()
return "Become a phoenix to strike the enemy."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .6
end

function GetDelay()
  return .4
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { FIR }
end

function SpellEffect()
  local damage = (4 * aMD * (aFIR/100) - dMR) * (100/dFIR)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/FlyingPhoenix.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class FlyingPhoenix : SpellScriptBase
    {
        public override string Name { get { return "Flying Phoenix"; } }
        public override string Description { get { return "Become a phoenix to strike the enemy."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.FIR, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 60; } }
        public override double Delay { get { return .4; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (4 * cMD * (cFIR/100) - tMR) * (100 / tFIR);
            DealDamage(damage);
        }
    }
}